### DISCUSSION TITLE
Message routing path

### DISCUSSION THREAD

**Discussion by zyra102388** [openthread/openthread#9998] [Q&A] [answered]
Hi. I am new to Thread. I want to ask how does the Thread network route a message. Please refer to the image attached.
Supposed End Device A will send a message to End Device B. How will the message be routed?
1. From End Device A -> Router 1 -> Router 2 -> End Device B
2. From End Device A -> Router 1 -> Thread Leader -> Router 2 -> End Device B

<img width="1164" alt="Thread" src="https://github.com/openthread/openthread/assets/139319638/d8225d1c-f0ee-4122-9138-4ac76ca4b837">

**Comment by jwhui** [comment, marked as answer]
Thread routers implement shortest-path any-to-any routing.

Assuming the link costs are all the same, packets from A to B will follow: A -> 1 -> 2 -> B
